dcm4che-core/src/main/java/org/dcm4che3/util/StreamUtils.java

```diff
@@ -150,8 +153,7 @@ public static void copy(InputStream in, OutputStream out, int len,
         }
     }
 
-    public static void copy(InputStream in, OutputStream out, int len,
-            int swapBytes) throws IOException {
+    public static void copy(InputStream in, OutputStream out, int len, int swapBytes) throws IOException {
```

Comment:
If you are going to tweak the signature, could you make that len a long instead?  Or perhaps automatically fix it if it is a -ve value?

Reply:
I have changed the code to pass long values.  I'm going to run some tests to make sure that it works properly.

Reply:
I have made the change locally so that a long is being passed and cast down to int internally, but I am concerned about side-effects.  There is no explicit StreamUtils coverage in DCM4CHE3 right now, and I think we should make a separate issue for handling that.  

I'm not really happy with StreamUtils in general to be honest, and we should fix it to handle not just large objects, but also to more stringently handle the InputStream being passed.  The code is consistently using read() == 0 as the end of file, which is not correct.  the InputStream is complete when read() == -1.

The main advantage this class has over the commons IOUtil is that it supports byte swapping.  Perhaps if we added an InputStream or OutputStream implementation that would do swapping inline we could just use the commons IO and really simplify this code...

Reply:
I actually think the best approach would be to write new methods which take long values and use IIS directly.  So for instance:

copy(ImageInputStream, OutputStream): long
copy(ImageInputStream, ImageOutputStream): long
copy(ImageInputStream, WritableChannel):long
copy(ImageInputStream, OutputStream, long offset, long length) 

etc...

The code is using IIS for all I/O, we should use the interface directly so that we get the most benefits.  We can either add those methods to StreamUtils directly, or even add a new StreamHandler interface that might allow us to intercept how copies are being done.  That would give us a nice way to override the buffer sizes or even add thread local buffering if required by an application.